smithy-typescript-ssdk-libs/server-node/src/index.spec.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { mkdtemp } from "fs/promises";
+import { createServer, IncomingMessage, request, RequestOptions, Server, ServerResponse } from "http";
+import * as os from "os";
+import * as path from "path";
+import { Readable } from "stream";
+
+import { convertRequest } from "./node";
+
+let socketPath: string;
+let promiseResolve: ([req, res]: [IncomingMessage, ServerResponse]) => void;
+
+let server: Server;
+beforeAll(async () => {
+  server = createServer(function (req, res) {
+    promiseResolve([req, res]);
+    resToEnd = res;
+  });
+  // Create a temporary named pipe where to run the server and obtain a request
+  socketPath = path.join(await mkdtemp(path.join(os.tmpdir(), "named-pipe-for-test-")), "server");
+  // TODO Add support to Windows by using '\\\\?\\pipe'
+  // See: https://nodejs.org/api/net.html#identifying-paths-for-ipc-connections
+  server.listen(socketPath);
+});
+
+let resToEnd: ServerResponse;
+
+function getRequest(options: RequestOptions & { body?: String }): Promise<[IncomingMessage, ServerResponse]> {
+  return new Promise((resolve, _) => {
+    promiseResolve = resolve;
+    request({
+      socketPath,
+      ...options,
+    }).end(Buffer.from(options.body || []));
+  });
+}
+
+afterAll(() => {
+  server?.close();
+});
+
+async function streamToString(stream: Readable) {
```

Comment:
@aws-sdk/node-http-handler streamCollector?

Reply:
Thanks for the suggestion. I see the following caveats:
* It returns a `Uint8Array` which I still need to convert to String.
* I'm introducing a new dependency `@aws-sdk/node-http-handler` only for this.

@kuhe what do you think?